### PR TITLE
setup: setup eslint and husky

### LINKED ISSUE #3: Setup eslint , husky

### PR DESCRIPTION
**Pull request by karansapolia**
fixes #3

### CODE REVIEW

**Review by nihalanisumit — APPROVED** [review]
LGTM

File: client/package.json [hunk 1]
```diff
@@ -13,9 +13,6 @@
     "test": "react-scripts test",
     "eject": "react-scripts eject"
   },
-  "eslintConfig": {
-    "extends": "react-app"
-  },
   "browserslist": {
```
Comment (piyushagwl007) [inline client/package.json:16]: Why did you remove it?

File: package.json [hunk 2]
```diff
@@ -9,11 +9,32 @@
...
+  },
+  "lint-staged": {
+    "*.{js,jsx}": [
+      "eslint"
+    ]
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged",
+      "pre-push": "lint-staged"
```
Comment (piyushagwl007) [inline package.json:37]: Can you explain lint-staged? in pre-push and pre-commit?

File: .eslintignore [hunk 3]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (piyushagwl007) [inline .eslintignore:1]: Will this ignore the subfolder node_modules as well?

**Review by piyushagwl007 — APPROVED** [review]
LGTM. Just need some explanation for some things :smiley:

File: .eslintignore (hunk 3 above)
Comment (karansapolia) [reply .eslintignore:1]: Yes. It will ignore all levels of nesting of mode_modules

File: package.json (hunk 2 above)
Comment (karansapolia) [reply package.json:37]: The code for the "lint-staged" key in package.json should be self-explanatory. It runs eslint on all staged js and jsx files. Read about the [lint-staged](https://github.com/okonet/lint-staged#configuration) package to learn more.

**Review by nimish-gupta — APPROVED** [review]
